bin/cylc-subscribe

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""cylc subscribe [OPTIONS] ARGS
+
+(This command is for internal use.)
+Invoke suite subscriber to receive published workflow output.
+"""
+
+import asyncio
+import json
+import sys
+if '--use-ssh' in sys.argv[1:]:
+    sys.argv.remove('--use-ssh')
+    from cylc.flow.remote import remrun
+    if remrun():
+        sys.exit(0)
+import time
+
+from google.protobuf.json_format import MessageToDict
+
+from cylc.flow.option_parsers import CylcOptionParser as COP
+from cylc.flow.network.scan import get_scan_items_from_fs, re_compile_filters
+from cylc.flow.network.subscriber import WorkflowSubscriber
+from cylc.flow.terminal import cli_function
+from cylc.flow.ws_messages_pb2 import PbWorkflow
+
+
+def print_workflow(topic, msg):
+    data = PbWorkflow()
+    data.ParseFromString(msg)
+    print('Received: ', topic.decode())
+    sys.stdout.write(
+        json.dumps(MessageToDict(data), indent=4) + '\n')
+
+
+def get_option_parser():
+    parser = COP(__doc__, comms=True, argdoc=[
+        ('REG', 'Suite name'),
+        ('[TOPIC]', 'Subscription topic to receive')])
+
+    return parser
+
+
+@cli_function(get_option_parser)
+def main(_, options, suite, topic=None):
+    host = None
+    port = None
+    cre_owner, cre_name = re_compile_filters(None, ['.*'])
+    while True:
```

Comment:
Random thought, without thinking too much, so can most likely be ignored/deleted :grimacing: If cylc-flow supported websockets, then we could do `cli -- websockets --> cylc-flow <-- uiserver -- web` I think? Probably with Python [websockets](https://websockets.readthedocs.io/en/stable/) module?

Reply:
Not sure if I follow completely (what do the arrows represent?), ZeroMQ is not exactly websockets... 
We can make conduits through one or the other (or joins between the websockets-ZeroMQ), and once we have GraphQL subscriptions at both WS and UIS then the CLI can use either for that 😃

Reply:
> If cylc-flow supported websockets

I raised this at Melbourne, it was voted down in favour of ZMQ.

Reply:
> > If cylc-flow supported websockets
> 
> I raised this at Melbourne, it was voted down in favour of ZMQ.

I think I raise the same too, but noted that we would need to create our own handlers for GraphQL... (although, it's doable (and I have already done it for REQ/REP))...

ZeroMQ is ok, and we can join it to UIS websockets if we needed..